test/notifications/display_test.dart

```diff
@@ -125,17 +145,16 @@ void main() {
 
       final messageStyleMessagesChecks =
         messageStyleMessages.mapIndexed((i, messageData) {
-          assert(messageData.realmUri == data.realmUri);
-          assert(messageData.userId == data.userId);
-
           final expectedSenderKey =
             '${messageData.realmUri}|${messageData.senderId}';
           final isLast = i == (messageStyleMessages.length - 1);
           return (Subject<Object?> it) => it.isA<MessagingStyleMessage>()
             ..text.equals(messageData.content)
             ..timestampMs.equals(messageData.time * 1000)
             ..person.which((it) => it.isNotNull()
-              ..iconBitmap.which((it) => isLast ? it.isNotNull() : it.isNull())
+              ..iconBitmap.which((it) => (isLast && expectedIconBitmap != null)
+                ? it.isNotNull().deepEquals(expectedIconBitmap)
+                : it.isNull())
```

Comment:
nit: This can be more compact while not being too long
```suggestion
              ..iconBitmap.which((it) => (isLast && expectedIconBitmap != null)
                ? it.isNotNull().deepEquals(expectedIconBitmap) : it.isNull())
```